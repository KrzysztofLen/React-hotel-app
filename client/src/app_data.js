var app_data = [
	{
		"id": 0,
		"name": "Eva",
		"hotel_name": "Sunset Willow Resort",
		"image": "https://preview.ibb.co/ivpH07/alex_block_354270_unsplash.jpg",
		"rate": 5,
		"more_activities": true,
		"activities_number": 6,
		"commented": 5,
		"peers": 46,
		"discussions": 29,
		"findings": 19,
		"questions": 10,
		"activity_level": 2
	},
	{
		"id": 1,
		"name": "Andrew",
		"hotel_name": "Mirror Hotel",
		"image": "https://preview.ibb.co/nbDc07/chuttersnap_221987_unsplash.jpg",
		"rate": 4,
		"more_activities": true,
		"activities_number": 3,
		"commented": 4,
		"peers": 15,
		"discussions": 46,
		"findings": 23,
		"questions": 77,
		"activity_level": 3
	},
	{
		"id": 2,
		"name": "Joseph",
		"hotel_name": "Ruby Garden Hotel",
		"image": "https://preview.ibb.co/gK1fL7/daniil_vnoutchkov_397278_unsplash.jpg",
		"rate": 2,
		"more_activities": false,
		"commented": 3,
		"is_new": true,
		"peers": 11,
		"discussions": 40,
		"findings": 35,
		"questions": 21,
		"activity_level": 1
	},
	{
		"id": 3,
		"name": "Eva",
		"hotel_name": "Fairyland Resort",
		"image": "https://preview.ibb.co/fmxLL7/rktkn_111844_unsplash.jpg",
		"rate": 5,
		"more_activities": true,
		"activities_number": 2,
		"commented": 4,
		"is_new": true,
		"peers": 74,
		"discussions": 23,
		"findings": 64,
		"questions": 2
	},
	{
		"id": 4,
		"name": "Andrew",
		"hotel_name": "Regal Spring Hotel",
		"image": "https://preview.ibb.co/eD6ann/shane_hauser_140922_unsplash.jpg",
		"rate": 1,
		"more_activities": false,
		"commented": 2,
		"is_new": true,
		"peers": 85,
		"discussions": 2,
		"findings": 1,
		"questions": 12
	},
	{
		"id": 5,
		"name": "Andrew",
		"hotel_name": "Sunny Park Hotel",
		"image": "https://preview.ibb.co/h5xgSn/scott_webb_67607_unsplash.jpg",
		"rate": 4,
		"more_activities": false,
		"commented": 4,
		"is_new": true,
		"peers": 44,
		"discussions": 33,
		"findings": 88,
		"questions": 123
	},
	{
		"id": 6,
		"name": "Marek",
		"hotel_name": "Modest Nugget Hotel",
		"image": "https://preview.ibb.co/bBxsYS/christoph_schulz_559688_unsplash.jpg",
		"rate": 3,
		"more_activities": true,
		"activities_number": 7,
		"commented": 4,
		"is_new": true,
		"peers": 41,
		"discussions": 32,
		"findings": 11,
		"questions": 33
	},
	{
		"id": 7,
		"name": "Arek",
		"hotel_name": "Light Shore Resort",
		"image": "https://preview.ibb.co/k237YS/park_troopers_221404_unsplash.jpg",
		"rate": 2,
		"more_activities": true,
		"activities_number": 4,
		"commented": 1,
		"is_new": true,
		"peers": 87,
		"discussions": 34,
		"findings": 73,
		"questions": 34
	},
	{
		"id": 8,
		"name": "Jarek",
		"hotel_name": "Crown Hotel",
		"image": "https://preview.ibb.co/eoD207/nichlas_andersen_225986_unsplash.jpg",
		"rate": 5,
		"more_activities": true,
		"activities_number": 91,
		"commented": 3,
		"is_new": true,
		"peers": 12,
		"discussions": 12,
		"findings": 5,
		"questions": 7
	},
	// New data schema. All data will be dynamic by adding to DB
	{
		"hotel_name": "Crown Hotel",
		"hotel_adress": {
			"adress": "Stare miasto",
			"city": "Wrocław",
			"province": "Dolnośląskie",
			"distance": 1.3
		},
		"description": "Lorem ipsum dolor sit amet",
		"hotel_stars": 5,
		"hotel_rating": 8.4, // Depends on that show "Good, very good, poor etc."
		"opinions": 831,
		"is_new": true,
		"is_apartment": true,
		"facilities": {
			"restaurant": true,
			"gym": false,
			"wifi": true,
			"card_payment": true,
			"game_room": false
		}
	}
];

export default app_data;