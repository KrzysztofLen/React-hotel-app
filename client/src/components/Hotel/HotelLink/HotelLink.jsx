// @flow
import React from 'react';
import {NavLink} from 'react-router-dom';

type Props = {
	id: number,
	hotelName: string
}

const HotelLink = (data: Props) => (
	<div className="hotel__details--name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
	</div>
);

export default HotelLink;