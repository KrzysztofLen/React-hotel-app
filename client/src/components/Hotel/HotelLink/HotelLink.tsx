import React from 'react';
import {NavLink} from 'react-router-dom';

interface IProps {
	id: number,
	hotelName: string
}

const HotelLink = (data: IProps) => (
	<div className="hotel__name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
	</div>
);

export default HotelLink;