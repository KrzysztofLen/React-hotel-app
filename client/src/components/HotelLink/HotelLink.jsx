import React from 'react';
import {NavLink} from 'react-router-dom';

const HotelLink = (data) => (
	<div className="hotel__details--name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
		<div className="hotel__details--adressBox">
			<span className="hotel__details--adress">{data.hotelAdress}</span>
			<p className="hotel__details--distance">(<span className="hotel__details--distanceValue">{data.hotelDistance}</span> km from center)</p>
		</div>
	</div>
);

export default HotelLink;