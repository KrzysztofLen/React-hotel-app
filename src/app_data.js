var app_data = [
	{
		"id": 0,
		"name": "Eva",
		"hotel_name": "Sunset Willow Resort",
		"image": "https://preview.ibb.co/ivpH07/alex_block_354270_unsplash.jpg",
		"rate": 5,
		"more_activities": true,
		"activities_number": 6,
		"commented": 5,
		"peers": 46,
		"discussions": 29,
		"findings": 19,
		"questions": 10,
		"activity_level": 2
	},
	{
		"id": 1,
		"name": "Andrew",
		"hotel_name": "Mirror Hotel",
		"image": "https://preview.ibb.co/nbDc07/chuttersnap_221987_unsplash.jpg",
		"rate": 4,
		"more_activities": true,
		"activities_number": 3,
		"commented": 4,
		"peers": 15,
		"discussions": 46,
		"findings": 23,
		"questions": 77,
		"activity_level": 3
	},
	{
		"id": 2,
		"name": "Joseph",
		"hotel_name": "Ruby Garden Hotel",
		"image": "https://preview.ibb.co/gK1fL7/daniil_vnoutchkov_397278_unsplash.jpg",
		"rate": 2,
		"more_activities": false,
		"commented": 3,
		"is_new": true,
		"peers": 11,
		"discussions": 40,
		"findings": 35,
		"questions": 21,
		"activity_level": 1
	},
	{
		"id": 3,
		"name": "Eva",
		"hotel_name": "Fairyland Resort",
		"image": "https://preview.ibb.co/fmxLL7/rktkn_111844_unsplash.jpg",
		"rate": 5,
		"more_activities": true,
		"activities_number": 2,
		"commented": 4,
		"is_new": true,
		"peers": 74,
		"discussions": 23,
		"findings": 64,
		"questions": 2
	},
	{
		"id": 4,
		"name": "Andrew",
		"hotel_name": "Regal Spring Hotel",
		"image": "https://preview.ibb.co/eD6ann/shane_hauser_140922_unsplash.jpg",
		"rate": 1,
		"more_activities": false,
		"commented": 2,
		"is_new": true,
		"peers": 85,
		"discussions": 2,
		"findings": 1,
		"questions": 12
	},
	{
		"id": 5,
		"name": "Andrew",
		"hotel_name": "Sunny Park Hotel",
		"image": "https://preview.ibb.co/h5xgSn/scott_webb_67607_unsplash.jpg",
		"rate": 4,
		"more_activities": false,
		"commented": 4,
		"is_new": true,
		"peers": 44,
		"discussions": 33,
		"findings": 88,
		"questions": 123
	},
	{
		"id": 6,
		"name": "Marek",
		"hotel_name": "Modest Nugget Hotel",
		"image": "https://preview.ibb.co/bBxsYS/christoph_schulz_559688_unsplash.jpg",
		"rate": 3,
		"more_activities": true,
		"activities_number": 7,
		"commented": 4,
		"is_new": true,
		"peers": 41,
		"discussions": 32,
		"findings": 11,
		"questions": 33
	},
	{
		"id": 7,
		"name": "Arek",
		"hotel_name": "Light Shore Resort",
		"image": "https://preview.ibb.co/k237YS/park_troopers_221404_unsplash.jpg",
		"rate": 2,
		"more_activities": true,
		"activities_number": 4,
		"commented": 1,
		"is_new": true,
		"peers": 87,
		"discussions": 34,
		"findings": 73,
		"questions": 34
	},
	{
		"id": 8,
		"name": "Jarek",
		"hotel_name": "Crown Hotel",
		"image": "https://preview.ibb.co/eoD207/nichlas_andersen_225986_unsplash.jpg",
		"rate": 5,
		"more_activities": true,
		"activities_number": 91,
		"commented": 3,
		"is_new": true,
		"peers": 12,
		"discussions": 12,
		"findings": 5,
		"questions": 7
	}
];

export default app_data;