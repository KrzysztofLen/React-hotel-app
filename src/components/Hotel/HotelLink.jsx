import React from 'react';
import {NavLink} from 'react-router-dom';

const HotelLink = (data) => (
	<div className="hotel__details--name">
		<NavLink to={"/hotel/" + data.id} className="hotel__link">{data.hotelName}</NavLink>
	</div>
);

export default HotelLink;