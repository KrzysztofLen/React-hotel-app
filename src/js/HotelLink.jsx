import React, {Component} from 'react';
import {NavLink} from 'react-router-dom';

const HotelLink = (data) => <NavLink to={"/question/" + data.id} className="question">{data.hotelName}</NavLink>;

export default HotelLink;