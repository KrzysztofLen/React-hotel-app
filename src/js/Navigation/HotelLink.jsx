import React from 'react';

export const HotelLink = () => (
	<li className="side-nav__item side-nav__item--active">
		<a href="#" className="side-nav__link">
			<span>Hotel's</span>
		</a>
	</li>
);