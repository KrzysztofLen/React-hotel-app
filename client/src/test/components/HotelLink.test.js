import React from 'react';
import {shallow} from 'enzyme';
import HotelLink from '../../components/Hotel/HotelLink/HotelLink';
import app_data from '../fixtures/app_data';

test('should render HotelRating Link correctly', () => {
	const wrapper = shallow(<HotelLink hotelName={app_data[0].hotel_name} id={app_data[0].id} />);
	expect(wrapper.debug()).toMatchSnapshot();
});