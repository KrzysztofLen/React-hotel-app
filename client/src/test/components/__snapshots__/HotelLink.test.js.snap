// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HotelRating Link correctly 1`] = `
<div
  className="hotel__details--name"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    ariaCurrent="true"
    className="hotel__link"
    to="/hotel/0"
  >
    Test Sunset Willow Resort
  </NavLink>
</div>
`;
